### PR TITLE
refactor(react-grid): improve group rows processing

### PR DESCRIPTION
**Pull request by kvet**

### CODE REVIEW

File: packages/dx-grid-core/src/plugins/editing-state/reducers.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ export const stopEditRows = (prevEditingRows, { rowIds }) => {
   return prevEditingRows.filter(id => !rowIdSet.has(id));
 };
 
-export const addRow = (addedRows, { row }) => [...addedRows, row];
```
Comment (dxbykov) [inline packages/dx-grid-core/src/plugins/editing-state/reducers.js]: Any reason for this change?

Comment (kvet) [reply packages/dx-grid-core/src/plugins/editing-state/reducers.js]: No. I will revert it

File: packages/dx-grid-core/src/plugins/grid-core/computeds.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+export const gridRows = data => data.map((row, index) => ({ row, defaultRowId: index }));
+
+export const gridRowIdGetter = getRowId => (gridRow) => {
+  if (getRowId) {
```
Comment (dxbykov) [inline packages/dx-grid-core/src/plugins/grid-core/computeds.js]: It should be resolved outside the result function. We don't have to check this on per-row call.

File: packages/dx-grid-core/src/plugins/table-group-row/computeds.js [hunk 3]
```diff
@@ -45,11 +46,15 @@ export const tableColumnsWithGrouping = (
 
 export const tableRowsWithGrouping = tableRows =>
   tableRows.map((tableRow) => {
-    if (tableRow.type !== TABLE_DATA_TYPE || tableRow.row.type !== 'groupRow') return tableRow;
+    if ((tableRow.type !== TABLE_UNKNOWN_TYPE && tableRow.type !== TABLE_DATA_TYPE)
+      || (tableRow.gridRow.type !== GRID_GROUP_TYPE && tableRow.gridRow.row.type !== 'groupRow')) {
```
Comment (gsobolev) [inline packages/dx-grid-core/src/plugins/table-group-row/computeds.js]: Should not the 'groupRow' string here be a constant?

File: packages/dx-react-grid/docs/reference/grid.md [hunk 4]
```diff
@@ -8,20 +8,16 @@ The Grid is a root container component designed to process and display data spec
 
 Name | Type | Default | Description
 -----|------|---------|------------
-rows | Array&lt;[Row](#row)&gt; | | Specifies the data the Grid displays.
-columns | Array&lt;[Column](#column)&gt; | | Specifies for which row object fields columns are created.
-getRowId | (row: [Row](#row)) => number &#124; string | null | Specifies the function used to get a row ID.
-getCellValue | (row: [Row](#row), columnName: string) => any | null | Specifies the function used to get a cell's value.
+data | Array&lt;any&gt; | | An array containing custom data. A user defines the access to this data. Refer to [Data Accessors](../guides/data-accessors.md) for details.
```
Comment (gsobolev) [inline packages/dx-react-grid/docs/reference/grid.md]: I believe it should be `rows`, not `data`.

File: packages/dx-react-grid/src/plugins/local-paging.test.jsx [hunk 5]
```diff
@@ -55,7 +56,7 @@ describe('LocalPaging', () => {
       .toBe(6);
   });
 
-  it('should paginate rows passed into based on the "currentPage" and "pageSize" getters', () => {
+  it('should paginatedRows rows passed into based on the "currentPage" and "pageSize" getters', () => {
```
Comment (SergeyAlexeev) [inline packages/dx-react-grid/src/plugins/local-paging.test.jsx]: Test name looks strange. `should paginatedRows` -> `should paginate` ?

**Pull request merged**